packages/web/src/components/admin_page/feature_switch_page/render_switch.jsx

```diff
@@ -0,0 +1,166 @@
+};
+
+const deleteSwitches = (switches, setSwitches) => switchToRemove => {
+    // eslint-disable-next-line no-unused-vars
+    const { [switchToRemove]: _, ...updatedSwitches } = { ...switches }; //creates a new object updatedSwitches that is a copy of switches with the siwtchToRemove, removed
+    setSwitches(updatedSwitches);
+};
+
+const RenderSwitches = () => {
+    const [selected, setSelected] = useState('');
+    const [showAddOverlay, setShowAddOverlay] = useState(false);
+    const [featureSwitches, setFeatureSwitches] = useContext(FsContext);
+    const [errorMessage, setErrorMessage] = useState(undefined);
+    const handleUpdateSwitches = updateSwitches(featureSwitches, setFeatureSwitches);
+    const handleDeleteSwitches = deleteSwitches(featureSwitches, setFeatureSwitches);
+
+    useEffect(() => {
+        axios(featureSwitchesURL, {
+            method: 'get',
+        }).catch(() => {
+            window.alert('API is not responding!');
+        });
+    }, []);
+
+    useEffect(() => {
+        if (selected) {
+            setShowAddOverlay(false);
+        }
+    }, [selected]);
+
+    useEffect(() => {
+        if (showAddOverlay) {
+            setSelected(undefined);
+        }
+    }, [showAddOverlay]);
+
+    const onUpdate = data => {
+        const switch_id = data.switch_id;
+        const is_enabled = data.is_enabled;
+
+        axios({
+            method: 'put',
+            url: parseToNewAPI(featureSwitchesURL),
+            headers: HEADERS,
+            data,
+        })
+            .then(() => {
+                if (errorMessage) {
+                    setErrorMessage(undefined);
+                }
+                setShowAddOverlay(false);
+                handleUpdateSwitches({ switch_id, is_enabled });
+            })
+            .catch(err => {
+                const message = err?.message;
+                setErrorMessage(message);
+            });
+    };
+
+    const renderSwitch = () => {
+        return (
+            <>
+                <OverlayTrigger show={showAddOverlay} placement="bottom" overlay={popover(onUpdate, errorMessage)}>
+                    <Button
+                        variant="primary"
+                        style={{ width: '100vw' }}
+                        onClick={() => {
+                            setShowAddOverlay(prev => !prev);
+                        }}>
+                        Create a Feature Switch
+                    </Button>
+                </OverlayTrigger>
+                <div className="switches-container">
+                    {Object.entries(featureSwitches).map(([switch_id, is_enabled]) => (
+                        <FeatureRow
+                            switch_id={switch_id}
+                            is_enabled={is_enabled}
+                            key={switch_id}
+                            selected={selected}
+                            setSelected={setSelected}
+                            setFeatureSwitches={setFeatureSwitches}
```

Comment:
Why do you pass setFeatureSwitches here? Don't you pass two more functions which already use setFeatureSwutches.